service/history/historyEngine.go

```diff
@@ -643,7 +643,11 @@ func (e *historyEngineImpl) NotifyNewTasks(
 		}
 
 		if len(tasksByCategory) > 0 {
-			e.queueProcessors[category].NotifyNewTasks(tasksByCategory)
+			processor, ok := e.queueProcessors[category]
+			if !ok {
+				continue
```

Comment:
if tasks are generated and there's no corresponding processor, I'd rather the system to panic instead of silently fail.